packages/sitecore-jss-nextjs/src/revalidate/revalidate-middleware.ts

```diff
@@ -0,0 +1,304 @@
+   * @returns {string} site name
+   */
+  protected getSiteName(pathname: string) {
+    let siteName = '';
+
+    const path = pathname.endsWith('/') ? pathname : pathname + '/';
+    const result = path.match('(.*?)\\/');
+
+    if (result && result[1] !== '') {
+      siteName = result[1];
+    }
+
+    return siteName;
+  }
+
+  /**
+   * Gets the path name from the full path
+   * @param {string} fullPath Full path
+   * @returns {string} path name
+   */
+  protected getPathName(fullPath: string) {
+    const pathParts = fullPath.split('/').filter((part) => part !== '');
+
+    if (pathParts.length >= 2) {
+      const siteName = `/${pathParts[0]}/`;
+      const path = `/${pathParts.slice(1).join('/')}`;
+      return path.startsWith(siteName) ? path.slice(siteName.length) : path;
+    }
+
+    return '/';
+  }
+
+  protected extractSiteName(path: string): string {
+    const siteName: string = path.split('/')[0];
+    return siteName;
+  }
+
+  /**
+   * Filters out the updated paths and language from the request body
+   * @param {NextApiRequest} req Next.js API request
+   * @returns {UpdatedPaths[]} updated paths
+   */
+  protected getFilteredUpdates(req: NextApiRequest): UpdatedPaths[] {
+    if (!req.body?.updates || this.isEmpty(req.body.updates)) {
+      return [];
+    }
+
+    return req.body?.updates
+      .filter(
+        (update: Entity) =>
+          update.entity_definition === EntityDefinition.LayoutData && update.entity_culture
+      )
+      .map((update: Entity) => {
+        return {
+          path: update.identifier,
+          language: update.entity_culture,
+        };
+      });
+  }
+
+  /**
+   * Gets the rewrite paths for the updated paths
+   * @param {PersonalizedResult[]} personalizeInfo Personalized results
+   * @returns {string[]} rewrite paths
+   */
+  protected getMultiSitePersonalizeRewrite(personalizeInfo: PersonalizedResult): string {
+    return `/${PERSONALIZE_PREFIX}${personalizeInfo.variantId}/${SITE_PREFIX}${personalizeInfo.path}`;
+  }
+
+  protected handleMultiSitePersonalization(
+    personalizeInfo: {
+      personalized: PersonalizedResult[];
+      nonPersonalized: {
+        path: string;
+      }[];
+    },
+    pathsToRevalidate: string[],
+    context: RevalidateMiddleware
+  ) {
+    const personalizedRewrite = personalizeInfo.personalized.map((info) => {
+      return context.getMultiSitePersonalizeRewrite(info);
+    });
+
+    const nonPersonalizedRewrite = personalizeInfo.nonPersonalized.map((info) => {
+      return getSiteRewrite(context.getPathName(info.path), {
+        siteName: context.getSiteName(info.path),
+      });
+    });
+
+    pathsToRevalidate.push(...personalizedRewrite, ...nonPersonalizedRewrite);
+  }
+
+  protected handleNonMultiSitePersonalization(
+    personalizeInfo: {
+      personalized: PersonalizedResult[];
+      nonPersonalized: {
+        path: string;
+      }[];
+    },
+    pathsToRevalidate: string[],
+    context: RevalidateMiddleware
```

Comment:
Why the need to pass `this` in to these functions as "context"?

Reply:
I passed it to make it look a bit cleaner but now passed the respective params.